src/dlsproto/node/neo/request/GetRange.d

```diff
@@ -149,77 +126,83 @@ public abstract scope class GetRangeProtocol_v1
 
     private Lzo lzo;
 
+    /// Indicator if the request is initialized well
+    bool initialised_ok;
+
     /***************************************************************************
 
-        Request handler. Reads the initial request args and starts the state
-        machine.
+
+        Called by the connection handler immediately after the request code and
+        version have been parsed from a message received over the connection.
+        Allows the request handler to process the remainder of the incoming
+        message, before the connection handler sends the supported code back to
+        the client.
+
+        Note: the initial payload is a slice of the connection's read buffer.
+        This means that when the request-on-conn fiber suspends, the contents of
+        the buffer (hence the slice) may change. It is thus *absolutely
+        essential* that this method does not suspend the fiber. (This precludes
+        all I/O operations on the connection.)
 
         Params:
-            connection = connection to client
-            msg_payload = initial message read from client to begin the request
-                (the request code and version are assumed to be extracted)
+            init_payload = initial message payload read from the connection
 
     ***************************************************************************/
 
-    final public void handle ( RequestOnConn connection, Const!(void)[] msg_payload )
+    public void preSupportedCodeSent ( Const!(void)[] msg_payload )
```

Comment:
It's recommended to use the new new request handler framework, which doesn't require these separate pre and post methods. See https://github.com/sociomantic-tsunami/swarm/blob/v5.x.x/src/swarm/neo/node/ConnectionHandler.d#L161.

Reply:
Err, but this hasn't been released yet - I've just tried to use it from swarm v5.0.x. Is it time for v5.1.0?

Reply:
Even with v5.1.0, I can't slip the swarm dependency upgrade in v14.x.x of dlsproto :-(.

Reply:
Ahh sorry, I forgot it wasn't in v5.0.x.